src/util/records.py

```diff
@@ -1202,6 +1202,9 @@ def delete_unused(cr, *xmlids, **kwargs):
     This function will remove records pointed by `xmlids` only if they are not referenced
     from any table.
 
+    For hierarchical records whose children are also removed (like product categories),
+    verify that those children are also not referenced.
```

Comment:
I'd suggest we make clearer that we check cascade removal of children, then we remove them all.
```suggestion
    from any table. For hierarchical records (like product categories), it verifies
    if the children marked as cascade removal are also not referenced. In which case
    the record and its children are all removed.
```